client/app/components/items-list/classes/StateStorage.js

```diff
@@ -0,0 +1,60 @@
+import { defaults } from 'lodash';
+import { clientConfig } from '@/services/auth';
+import { $location } from '@/services/ng';
+import { parse as parseOrderBy, compile as compileOrderBy } from './Sorter';
+
+export class StateStorage {
+  constructor(state = {}) {
+    this._state = { ...state };
+  }
+
+  getState() {
+    return defaults(this._state, {
+      page: 1,
+      itemsPerPage: clientConfig.pageSize,
+      orderByField: 'created_at',
+      orderByReverse: false,
+      searchTerm: '',
+      tags: [],
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  setState() {
+  }
+}
+
+export class UrlStateStorage extends StateStorage {
+  getState() {
+    const defaultState = super.getState();
+    const params = $location.search();
+
+    const searchTerm = params.q || '';
+
+    // in search mode order by should be explicitly specified in url, otherwise use default
+    const defaultOrderBy = searchTerm !== '' ? '' : compileOrderBy(defaultState.orderByField, defaultState.orderByReverse);
```

Comment:
If `searchTerm !== ''` it's gonna anyway be taken from `params.order` so this is redundant, no?

If not, consider running this only when necessary (within the `param.order` condition).

Reply:
1. it's not redundant - it's used as default value for `params.order` few lines below;
2. it's running only when necessary (also, see few lines below).

If you mean adding condition on `if (params.order)` - IMO this code looks better (more linear); adding a condition makes sense, but this function is called very rarely (only on page load), so I see no profit from it.